AsyncDisplayKit/Private/ASWeakMap.h

```diff
@@ -0,0 +1,59 @@
+//
+//  ASWeakMap.h
+//  AsyncDisplayKit
+//
+//  Created by Chris Danford on 7/11/16.
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+/**
+ * This class is used in conjunction with ASWeakMap.  Instances of this type are returned by an ASWeakMap,
+ * must retain this value for as long as they want the entry to exist in the map.
+ */
+@interface ASWeakMapEntry<Value> : NSObject
+
+@property (nonatomic, retain, readonly) Value value;
```

Comment:
Can we use the word `strong` here? IIRC `retain` as a property attribute is a holdover from MRR days.